csrc/device_lower/validation.cpp

```diff
@@ -598,7 +598,25 @@ void validateAndCollectVectorizeInfo(Fusion* fusion) {
             "Only allow misaligned vectorization between global and local memory.");
         has_misaligned_vectorize_dim = true;
       }
+
+      // ParallelType::Group is used for both reduction & normalization.
```

Comment:
Is this really necessary? Shared memory is always aligned. If the producer is in global memory, then that should be a temporary work buffer, so it should be always a contiguous, aligned buffer.

Reply:
I mean between register and shared memory, so we need to ensure the registers are aligned when doing vectorized read/write.

Reply:
In this case, `T13 = GroupedReductionOf(T14)`, `T14 (registers)` should be aligned becase runtime funciton uses vectorized copy from `T14(registers)` to shared memory.
```
T13_l_float[ iblockIdx.x68{( ceilDiv(( ceilDiv(( ceilDiv(i2, 4) ), blockDim.x) ), 1) )}, ithreadIdx.x67{blockDim.x}, iUS69{1}, iG65{4}, rthreadIdx.y62{blockDim.y} ] ca_pos( 3 ) produce_pos( 2 )
   = reduction( T14_l_float[ iblockIdx.x54{( ceilDiv(( ceilDiv(( ceilDiv(i2, 4) ), blockDim.x) ), 1) )}, ithreadIdx.x53{blockDim.x}, rS60{( ceilDiv(( ceilDiv(( ceilDiv(i1, blockDim.y) ), 2) ), 1) )}rf, iUS55{1}, iS51{4}, ithreadIdx.y57{blockDim.y}rf, rUS61{1}rf, rS59{2}rf ] ca_pos( 2 ) produce_pos( 8 ), op = add, initial value = float(0), allreduce = false )
```

Reply:
I think I got it. The comment seems wrong, though.

```
since they are register arrays defined in runtime function
```

This `producer_tv` is not defined in the runtime functions, right?

Reply:
That's right. `producer_tv` is not defined in the runtime functions, it is just passed to the runtime function.

Reply:
comment is revised as 
```
      // ParallelType::Group is used for both reduction and normalization.
      // In grouped outer reduction, the runtime function uses vectorized data
      // transfers between registers and shared memory. The producer tensor is
      // stored in registers and loaded into shared memory in a vectorized
      // manner, so we add it to the vectorizedAccesses map to ensure register
      // alignment.
```